.github/workflows/build.yml

```diff
@@ -0,0 +1,170 @@
+name: Build Artifacts
+
+on:
+  workflow_call:
+    inputs:
+      version:
+        required: true
+        type: string
+      unix:
+        default: true
+        type: boolean
+      windows32:
+        default: true
+        type: boolean
+
+  workflow_dispatch:
+    inputs:
+      version:
+        description: |
+          VERSION: yyyy.mm.dd[.rev] or rev
+        required: true
+        type: string
+      unix:
+        description: youtube-dl, youtube-dl.tar.gz
+        default: true
+        type: boolean
+      windows32:
+        description: youtube-dl.exe
+        default: true
+        type: boolean
+
+permissions:
+  contents: read
+
+jobs:
+  unix:
+    if: inputs.unix
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Needed for changelog
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.9"
+
+      - name: Install Requirements
+        run: |
+          sudo apt -y install zip pandoc man sed
+
+      - name: Prepare
+        run: |
+          python devscripts/update_version.py "${{ inputs.version }}"
+          python devscripts/changelog.py --update
+          python devscripts/make_lazy_extractors.py youtube_dl/extractor/lazy_extractors.py
+
+      - name: Build Unix platform-independent binary
+        run: |
+          make all tar
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-bin-${{ github.job }}
+          path: |
+            youtube-dl
+            youtube-dl.tar.gz
+          compression-level: 0
+
+  windows32:
+    if: inputs.windows32
+    runs-on: windows-2022
+    env:
+      PYCRYPTO: pycrypto-2.6.1-cp34-none-win32
+      # Temporary workaround for Python 3.4/5 failures - May 2024
+      PIP_TRUSTED_HOST: "pypi.python.org pypi.org files.pythonhosted.org"
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 3.4
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.4"
```

Comment:
This is used bc thats what the old youtub-dl executable used. Mayhaps it would make sense to bump this eventually?

Reply:
As long as 3.4 is supported in the codebase, it can/should stay at this value, because it means that the bundled .exe runs on Windows XP.

At one time there was a [graphic that explained this in Lesmiscore's PR workflow](https://github.com/ytdl-org/youtube-dl/pull/30644/files#diff-5c3fa597431eda03ac3339ae6bf7f05e1a50d6fc7333679ec38e21b337cb6721
>    # reason to choose 3.4: https://media.discordapp.net/attachments/807245652072857613/942409077701619742/unknown.png

but owing to Discordance this has been discarded.

The workflow gets or prepares some antique wheels to provide crypto support for this Python version.

Reply:
The graphic does not explain anything. It just shows a binary running 3.4. If you desire to support 3.4 forever then i guess it will have to stay this way

Reply:
Apparently Windows XP is quite popular in some parts of the world. As long as we're supporting 2.7, 3.4 is not much of a stretch.